File: Tp Soap/Q1EnJava/src/Reserve/RechercheHotel.java
package Reserve;

import java.util.ArrayList;
import java.util.Date;
import Reserve.Chambre;

import Reserve.Adresse;

public class RechercheHotel {
	
	private ArrayList<Hotel> list = null;

	public RechercheHotel() {
		
		this.list = new ArrayList();
	}
	public Boolean add(Hotel hotel){
		
		if(!list.contains(hotel)){
			list.add(hotel);
			return true;
		}else {
			return false;
		}
	}
	public ArrayList rechercherHotel(String villeSejour , Date datedepart , Date dateArrivee , Double prixMin , Double prixMax , int nombreEtoile , int nombrePersonne)
    {
        ArrayList hotels = new ArrayList();
        for(Hotel  h:  list)
        {

        	if (h.getAdresse().getVille().equals(villeSejour) && h.nombreEtoile==nombreEtoile){
        		for (Chambre ch : h.chambres) {
        		
        			if (dateArrivee.compareTo(ch.getDateDebutDisponibilite()) < 0 && datedepart.compareTo(ch.getDateFinDisponibilite()) > 0 && ch.getTypeChambre().getPrixBase() <= prixMax && ch.getTypeChambre().getPrixBase() >= prixMin && ch.getTypeChambre().getNbrLits() == nombrePersonne) {
        			hotels.add(h);
        			}
				}
        		
        	}
        }
        return hotels ;
    }


    public ArrayList getHotels()
    {
        return list;
    }

}

File: Tp Soap/Q1/Q1/RechercheHotel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Collections;

namespace Q1
{
    class RechercheHotel
    {
        private ArrayList list = null;

        public RechercheHotel()
        {
            this.list = new ArrayList();
        }

        public Boolean add(Hotel hotel)
        {
            if (!list.Contains(hotel))
            {
                list.Add(hotel);
                return true;
            }
            else
                return false;
        }


        public ArrayList rechercherHotel(String villeSéjour, DateTime datedepart, DateTime dateArrivée, Double prixMin, Double prixMax, int nombreEtoile, int nombrePersonne)
        {
            ArrayList hotels = new ArrayList();
            foreach (Hotel h in list)
            {

                if (h.Adresse.Ville.Equals(villeSéjour) && h.NombreEtoile == nombreEtoile)
                {

                    foreach (Chambre c in h.Chambres)
                    {

                        int result1 = DateTime.Compare(dateArrivée, c.DateDebutDispo);
                        int result2 = DateTime.Compare(datedepart, c.DateFinDispo);
                        if (result1 >= 0 && result2 <= 0 && c.TypeChambre.PrixBase <= prixMax && c.TypeChambre.PrixBase >= prixMin && c.TypeChambre.NbrLits == nombrePersonne)

                        {

                            hotels.Add(h); break;
                        }
                    }

                }
            }
            return hotels;
        }


        public ArrayList getHotels()
        {
            return list;
        }



    }
}
